recipes/zeus_expected/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get
+from conan.tools.cmake import cmake_layout
+import os
+
+required_conan_version = ">=2.0"
+
+
+class ZeusExpectedConan(ConanFile):
+    name = "zeus_expected"
+
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/zeus-cpp/expected"
+    description = "Backporting std::expected to C++17."
+    topics = ("cpp17", "expected")
+    license = "MIT"
+
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    no_copy_source = True
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 17)
```

Comment:
```suggestion
    @property
    def _min_cppstd(self):
        return 17

    @property
    def _minimum_compilers_version(self):
        return {
            "gcc": "7",
            "clang": "5",
            "Visual Studio": "15.7",
            "apple-clang": "10",
        }


    def validate(self):
        if self.settings.compiler.get_safe("cppstd"):
            check_min_cppstd(self, self._min_cppstd)

        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
        if min_version and Version(self.settings.compiler.version) < min_version:
            raise ConanInvalidConfiguration(f"{self.name} requires C++{self._min_cppstd}, which your compiler does not support.")
```

Need to check compiler version when not having settings.compiler.cppstd defined (Conan 1.x)

Reply:
Thanks for the detailed example. As a C++ library, I haven't consider that before. Is it only need for conan 1.x? If so, I would leave a comment so that we can remove those boilerplate code later.

Reply:
Don't need to comment because is well used over all ConanCenterIndex and part of templates: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py

In the future, after dropping Conan 1.x, will be able to track it anyway and remove. Thank you for asking.